### PR TITLE
feat: added balancing & normalization data in main.py, developped the LogisticRegression model

### PR DESCRIPTION
**Pull request by SaFouNaldo**
Premièrement, j'ai tout décaler (pas important) mais c'est juste pour qu'on puisse run.
Deuxièment, dans `main.py` j'ai rajouter la normalization et le "balancing" si vous voulez l'utiliser
Enfin, dans `methods/logistic_regression.py` j'ai ecris le code pour le model. Hésitez pas à lire ce que j'ai fait et demander des questions.

Pour run il faut : 

1.  Déplacer le `Data_MS1_2025/features.npz` dans le `sciper1_sciper2_sciper3_project` (i.e au mm lvl que main.py)
2.  Run la commande suivant sans le balancing : `python main.py --method logistic_regression --lr 0.01 --max_iters 100` Ou run la commande suivante avec le balancing: `python main.py --method logistic_regression --lr 0.01 --max_iters 100 --balance_data`

En faisant ça on obtient les résultats suivants : 
image
Pour les valeurs qu'on obtient pour le balanced vous pouvez changer les hyperparametres pour obtenir de meilleurs performances.

Vous pouvez tester le code en faisant 
1. git clone ... (pas la peine si deja fait)
2. `git fetch`
3. `git checkout skalliBranch`
4. executez les commandes précédentes

### CODE REVIEW

**Comment by SaFouNaldo** [comment]
`python main.py --method logistic_regression --lr 0.001 --max_iters 1000000` 
--> On obtient :
Train set: accuracy = 66.667% - F1-score = 0.535209
Test set:  accuracy = 58.333% - F1-score = 0.344396
`python main.py --method logistic_regression --lr 0.003 --max_iters 100000`
--> on obtient : 
Train set: accuracy = 65.608% - F1-score = 0.461014
Test set:  accuracy = 58.333% - F1-score = 0.322455
`python main.py --method logistic_regression --lr 0.002 --max_iters 100000`
--> On obtient : 
Train set: accuracy = 65.608% - F1-score = 0.460848
Test set:  accuracy = 60.417% - F1-score = 0.332308
Largement meilleur que ceux des profs

**Review by adidix — APPROVED** [review]

**Review by AKallala — APPROVED** [review]
